packages/@aws-cdk/aws-servicecatalogappregistry/README.md

```diff
@@ -65,33 +65,39 @@ const importedApplication = appreg.Application.fromApplicationArn(
 );
 ```
 
-## Automatic-Application
+## Register-Application
 
-An AppRegistry L2 construct to automatically create an application with the given name and description.
-The application name must be unique at the account level and it's immutable.
-`AutomaticApplication` L2 construct will automatically associate all stacks in the given scope, however
-in case of a `Pipeline` stack, stage underneath the pipeline will not automatically be associated and
-needs to be associated separately.
-If cross account stack is detected, then this construct will automatically share the application to consumer accounts.
-Cross account feature will only work for non environment agnostic stacks.
-
-Following will create an Application named `MyAutoApplication` in account `123456789012` and region `us-east-1`
-and will associate all stacks in the `App` scope to `MyAutoApplication`.
+If you want to create an Application named `MyRegisteredApplication` in account `123456789012` and region `us-east-1`
+and want to associate all stacks in the `App` scope to `MyRegisteredApplication`, then use as shown in the example below:
 
 ```ts
 const app = new App();
-const autoApp = new appreg.AutomaticApplication(app, 'AutoApplication', {
-    applicationName: 'MyAutoApplication',
-    description: 'Testing auto application',
+const registeredApp = new appreg.RegisterApplication(app, 'RegisterApplication', {
```

Comment:
Hi, I am sure the team thought deeply about a name of this class. Still, I think having a class name expressing an action instead of an object looks unintuitive. Perhaps that should be a factory method on the application class? Something like `appreg.Application.registerStacks(app)`?

Related topics in AWS Construct Library Design Guidelines:
* [Construct class](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#construct-class) - _"The name of resource constructs must be identical to the name of the resource in the AWS API, which should be consistent with the resource name in the AWS CloudFormation spec"_
* [Factories](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#factories)

P.S.: Another reference can be [cdk-nag](https://github.com/cdklabs/cdk-nag) project.